source/service.go

```diff
@@ -237,7 +237,22 @@ func (sc *serviceSource) extractHeadlessEndpoints(svc *v1.Service, hostname stri
 			headlessDomains = append(headlessDomains, fmt.Sprintf("%s.%s", v.Spec.Hostname, hostname))
 		}
 		for _, headlessDomain := range headlessDomains {
-			if sc.publishHostIP == true {
+			if sc.publishHostIP && externalIp {
+				node, err := sc.nodeInformer.Lister().Get(v.Spec.NodeName)
+				if err != nil {
+					return nil
+				}
+				if v.Status.Phase == v1.PodRunning {
+					for _, x := range node.Status.Addresses {
+						if x.Type == v1.NodeExternalIP {
+							log.Debugf("Generating matching endpoint %s with External %s", headlessDomain, v.Status.HostIP)
```

Comment:
ExternalIP should be `x.Address`,  `v.Status.HostIP` is internal

Reply:
thanks! fixed, had it wrong in the logging 👍